src/question/tests.py

```diff
@@ -1,3 +1,238 @@
+from django.contrib.auth.models import User
 
-# Create your tests here.
+from account.models import Profile
+from common.models import Votes
+from .models import Question, Answer, Comment
+
+class OneVoterQuestion(APITestCase):
+    model = Question
+    url_name = 'questions'
+
+    abstract_comment = None
+    upvoters = None
+    downvoters = None
+    url = None
+    voter = None
+
+
+    def prepare(self):
+        self.abstract_comment = self.model.objects.all()[0]
+        abstract_comment_url = ''.join(['http://localhost:8000/api/v1/', self.url_name, '/'])
+        self.upvoters = self.abstract_comment.votes.upvoters
+        self.downvoters = self.abstract_comment.votes.downvoters
+        self.url = ''.join([abstract_comment_url, str(self.abstract_comment.id), '/'])
+        self.voter = User.objects.get(username='voter')
+
+
+    def setUp(self):
+        # Users
+        submitter = Profile.objects.create(
+            user=User.objects.create(
+                username='submitter'
+            )
+        )
+        voter = Profile.objects.create(
+            user=User.objects.create(
+                username='voter',
+            )
+        )
+        # Content
+        question = Question.objects.create(
+            title='What is love?',
+            text='Baby don\'t hurt me!',
+            votes=Votes.objects.create(),
+            show_username=True,
+            owner=submitter
+        )
+        answer = Answer.objects.create(
+            text='I like apples.',
+            votes=Votes.objects.create(),
+            show_username=True,
+            owner=submitter,
+            parent=question,
+            is_accepted=False
+        )
+        comment = Comment.objects.create(
+            text='Apples suck.',
+            votes=Votes.objects.create(),
+            show_username=True,
+            owner=submitter,
+            parent_answer=answer
+        )
+
+
+    def test_zero_votes(self):
+        self.prepare()
+        self.assertEqual(self.upvoters.all().count(), 0)
+        self.assertEqual(self.downvoters.all().count(), 0)
+
+
+    def test_up_up(self):
+        self.prepare()
+        self.client.force_login(self.voter)
+        response = self.client.put(self.url, {'user_vote': 'up'})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(self.upvoters.all().count(), 1)
+        self.assertEqual(self.downvoters.all().count(), 0)
+        self.assertEqual(self.upvoters.filter(user=self.voter).count(), 1)
+        self.assertEqual(self.downvoters.filter(user=self.voter).count(), 0)
+
+        response = self.client.put(self.url, {'user_vote': 'up'})
+        self.client.logout()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(self.upvoters.all().count(), 0)
+        self.assertEqual(self.downvoters.all().count(), 0)
+        self.assertEqual(self.upvoters.filter(user=self.voter).count(), 0)
```

Comment:
I think it should be the client's job to cancel votes and only send the final user vote to the server: up, down, or none.

Reply:
Moved voting to https://github.com/DevTeamSCH/vikoverflow-backend/pull/12

Reply:
Also, fixed